cmd/disk-cache-fs.go

```diff
@@ -54,28 +59,22 @@ type cacheFSObjects struct {
 // Inits the cache directory if it is not init'ed already.
 // Initializing implies creation of new FS Object layer.
 func newCacheFSObjects(dir string, expiry int, maxDiskUsagePct int) (*cacheFSObjects, error) {
-	cachefsPath := pathJoin(dir, cacheDataDir)
-	obj, err := NewFSObjectLayer(cachefsPath)
+	obj, err := NewFSObjectLayer(dir)
 	if err != nil {
 		return nil, err
 	}
-	expiredBucketsPath := pathJoin(dir, cacheTrashDir)
-	if err := os.MkdirAll(expiredBucketsPath, 0777); err != nil {
-		return nil, err
-	}
-	expiredMultipartBucketsPath := pathJoin(dir, cacheTrashDir, minioMetaMultipartBucket)
-	if err := os.MkdirAll(expiredMultipartBucketsPath, 0777); err != nil {
-		return nil, err
-	}
-	expiredMetadataBucketPath := pathJoin(dir, cacheTrashDir, minioMetaBucket, bucketMetaPrefix)
-	if err := os.MkdirAll(expiredMetadataBucketPath, 0777); err != nil {
+
+	trashPath := pathJoin(dir, minioMetaBucket, cacheTrashDir)
+	if err := os.MkdirAll(trashPath, 0777); err != nil {
 		return nil, err
 	}
+
 	if expiry == 0 {
 		expiry = globalCacheExpiry
 	}
 	var cacheFS cacheFSObjects
 	fsObjects := obj.(*FSObjects)
+	fsObjects.metaJSONFile = cacheMetaJSONFile
```

Comment:
We should change the `.metaJSONFile` such that we take this an input for the NewFSObjects() and avoid doing this kind of change.

Reply:
used your suggestion.